plugin-gradle/src/main/java/com/diffplug/gradle/spotless/FormatExtension.java

```diff
@@ -623,12 +624,46 @@ public EclipseWtpConfig eclipseWtp(EclipseWtpFormatterStep type, String version)
 		return new EclipseWtpConfig(type, version);
 	}
 
+	/**
+	 * Given a regex with *exactly one capturing group*, disables formatting
+	 * inside that captured group.
+	 */
+	public void toggleOffOnRegex(String regex) {
+		this.togglePair = PipeStepPair.named(PipeStepPair.defaultToggleName()).regex(regex).buildPair();
+	}
```

Comment:
Alternatively, we could expand the comment with a sentence like,

> `If the regex has more than one capturing group, the behavior of this method is _undefined_.`

Reply:
or `... _unspecified_.`

Reply:
I had the same thought!  So far as I could tell, there is no way to count the groups in a regex.  You *can* count the groups in a valid match ([`Matcher::groupCount()`](https://docs.oracle.com/javase/7/docs/api/java/util/regex/Matcher.html#groupCount())), but you need a valid match before you can do that.  So we *could* check it at runtime, but I think that gives a false sense of security, because it only checks *if* the regex actually matches.  The vast majority of the time the regex will not match, so you're getting the error very very late.

Since we can't check the arg at "arg-time", we're stuck with just being very stern in the javadoc.  If their regex has no capturing groups, then if it eventually matches some file then they'll get a runtime error (no better or worse than we could give with `groupCount()`).  If their regex has two or more capturing groups, we just always use the first group, and ignore the rest.  So I don't think it's undefined, it's just that it only makes sense to have exactly one capturing group, since it's the only one that does anything.  I'd be happy to merge a PR which makes the wording more explicit about the errors and behavior you get if you break the rules, but I think we get the best SnR by just saying "regex with exactly one capturing group".